packages/components/src/scroll-lock/index.js

```diff
@@ -1,114 +1,64 @@
 /**
  * WordPress dependencies
  */
-import { Component } from '@wordpress/element';
+import { useEffect } from '@wordpress/element';
+
+let previousScrollTop = 0;
 
 /**
- * Creates a ScrollLock component bound to the specified document.
- *
- * This function creates a ScrollLock component for the specified document
- * and is exposed so we can create an isolated component for unit testing.
- *
- * @param {Object} args Keyword args.
- * @param {HTMLDocument} args.htmlDocument The document to lock the scroll for.
- * @param {string} args.className The name of the class used to lock scrolling.
- * @return {WPComponent} The bound ScrollLock component.
+ * @param {boolean} locked
+ * @param {string} className
  */
-export function createScrollLockComponent( {
-	htmlDocument,
-	className = 'lockscroll',
-} = {} ) {
-	if ( ! htmlDocument && typeof document !== 'undefined' ) {
-		htmlDocument = document;
-	}
-
-	let lockCounter = 0;
-
-	/*
-	 * Setting `overflow: hidden` on html and body elements resets body scroll in iOS.
-	 * Save scroll top so we can restore it after locking scroll.
-	 *
-	 * NOTE: It would be cleaner and possibly safer to find a localized solution such
-	 * as preventing default on certain touchmove events.
-	 */
-	let previousScrollTop = 0;
-
-	/**
-	 * Locks and unlocks scroll depending on the boolean argument.
-	 *
-	 * @param {boolean} locked Whether or not scroll should be locked.
-	 */
-	function setLocked( locked ) {
-		const scrollingElement =
-			htmlDocument.scrollingElement || htmlDocument.body;
+function setLocked( locked, className ) {
+	const scrollingElement = document.scrollingElement || document.body;
```

Comment:
It's usually a good practice to get the document from an element in the DOM (like `element.ownerDocument`) instead of referencing it directly like this, so we avoid issues with iframes. Not a blocker to this PR as this is also happening on trunk. Just noting here in case this becomes a problem in the future.

Reply:
Hmm, that's interesting and not easy to fix without API change. The `ScrollLock` component doesn't have access to any element. It doesn't render anything it could attach a ref to. A custom document would need to be a prop.

If I understand the issue correctly, it would happen if a script running in one frame renders a React UI in another frame:
```js
ReactDOM.render( <App />, document.getElementById( 'frame' ).contentDocument.getElementById( 'root' ) );
```
React itself always gets the right document (`mountRoot.ownerDocument`) but our components don't.

Reply:
I guess `ScrollLock` could return an empty element and pass its ref to this function. This may break styles with `:first-child` and similar though.